src/content/en/tools/workbox/v3/guides/precache-files.md

```diff
@@ -0,0 +1,296 @@
+
+Running `workbox wizard` will step through a set of questions to understand
+your project. Below is an example set of questions.
+
+![Workbox CLI Steps](../images/guides/get-started/cli-wizard-5.png)
+
+After you've run the wizard we'll need to add the location of our service
+worker to the config file. Notice the `swSrc` entry added to the workbox
+config.
+
+```javascript
+module.exports = {
+  "globDirectory": "app/",
+  "globPatterns": [
+    "**/*.{html,ico,svg,png,js,css}"
+  ],
+  "swDest": "dist/sw.js",
+  "swSrc": "app/sw.js"
+};
+```
+
+With this, you can run the `injectManifest` command that will read the file
+defined in `swSrc` and generate a version of it with the manifest to the
+location defined in `swDest`.
+
+In this case it will take `app/sw.js`, inject the manifest and write the file
+to `dist/sw.js`.
+
+When the inject manifest command is run, it'll print up some stats on the
+number of files and the total size.
+
+```
+$ workbox injectManifest workbox-config.js
+The service worker was written to dist/sw.js
+11 files will be precached, totalling 43.5 kB.
+```
+
+Run the `workbox injectManifest` command before you deploy your site to ensure
+the precache list is up to date.
+
+### Using workbox-build
+
+The `workbox-build` module provides a method to inject the precace manifest
+into your service worker.
+
+This module is the best option if you currently have a Node based build process
+for your project (i.e. you are using a tool like [Gulp](https://gulpjs.com/)).
+
+To install the module just run:
+
+```
+npm install workbox-build --save-dev
+```
+
+Then in your build process you'll want to `require` in the module and
+use the
+[injectManifest method](../reference-docs/latest/module-workbox-build#.injectManifest)
+to read in your service worker and output a version with the injected manifest.
+
+```javascript
+const workboxBuild = require('workbox-build');
+
+workboxBuild.injectManifest({
+  swSrc: path.join(__dirname, 'app', 'sw.js'),
+  swDest: path.join(__dirname, 'dist', 'sw.js'),
+  injectionPointRegexp: /(\.precacheAndRoute\()\s*\[\s*\]\s*(\))/,
+})
+.catch((err) => {
+  console.error(`Unable to inject the precache manifest into sw.js`);
+  throw err;
+});
+```
+
+Running this will read in `app/sw.js` and output a version of it to `dist/sw.js`
+with the manifest injected into it.
+
+To use this in Gulp, you would do the following:
+
+```javascript
+const workboxBuild = require('workbox-build');
+const gulp = require('gulp');
+
+gulp.task('build-sw', () => {
+  return workboxBuild.injectManifest({
+    swSrc: path.join(__dirname, 'app', 'sw.js'),
+    swDest: path.join(__dirname, 'dist', 'sw.js'),
+    injectionPointRegexp: /(\.precacheAndRoute\()\s*\[\s*\]\s*(\))/,
+  })
+  .catch((err) => {
+    console.error(`Unable to inject the precache manifest into sw.js`);
+    throw err;
+  });
+});
+```
+
+### Using the webpack Plugin
+
+TODO: How to inject manifest
```

Comment:
Need to fill out these sections or remove them

Reply:
Remove them.